### PR TITLE
planner: support record tiflash plan in general log

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:
support record tiflash plan in general log

### What is changed and how it works?
After the `postOptimize`, we should 

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
```
[2022/11/29 15:53:58.504 +08:00] [INFO] [optimizer.go:298] [GENERAL_LOG] [SQL="select count(*) from t"] [TiFlashTables="[t]"]
[2022/11/29 15:54:08.525 +08:00] [INFO] [optimizer.go:298] [GENERAL_LOG] [SQL="select count(*) from t join t t1"] [TiFlashTables="[t,t]"]
[2022/11/29 15:55:12.180 +08:00] [INFO] [optimizer.go:298] [GENERAL_LOG] [SQL="explain select count(*) from t join t1 on t.a = t1.a"] [TiFlashTables="[t1,t]"]
```
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/optimizer.go [hunk 1]
```diff
@@ -301,6 +310,35 @@ func DoOptimize(ctx context.Context, sctx sessionctx.Context, flag uint64, logic
 	return finalPlan, cost, nil
 }
 
+func hasTiFlashPlan(p PhysicalPlan) (ok bool, tables []string) {
```
Comment (qw4990) [inline planner/core/optimizer.go:313]: Can we use the existing `useTiFlash` function here?

**Review by qw4990 — APPROVED** [review]

**Review by chrysan — APPROVED** [review]

**Pull request merged**